scripts/VCSA/upgrade.sh

```diff
@@ -101,32 +100,29 @@ if [[ ${VIC_UI_HOST_URL: -1: 1} != "/" ]] ; then
 fi
 
 prompt_thumbprint_verification() {
-    read -p "Are you sure you trust the authenticity of this host (yes/no)? " SHOULD_ACCEPT_VC_FINGERPRINT
-    if [[ ! $(echo $SHOULD_ACCEPT_VC_FINGERPRINT | grep -woi "yes\|no") ]] ; then
-        echo Please answer either \"yes\" or \"no\"
-        prompt_thumbprint_verification
-        return
-    fi
-
-    if [[ $SHOULD_ACCEPT_VC_FINGERPRINT = "no" ]] ; then
-        read -p "Enter SHA-1 thumbprint of target VC: " VC_THUMBPRINT
-    fi
+    while true; do
+        IFS="" read -r -p "Are you sure you trust the authenticity of this host (yes/no)? " SHOULD_ACCEPT_VC_FINGERPRINT
+        case "${SHOULD_ACCEPT_VC_FINGERPRINT}" in
+            [Yy][Ee][Ss] ) return;;
+            [Nn][Oo] ) IFS="" read -r -p "Enter SHA-1 thumbprint of target VC: " VC_THUMBPRINT; return;;
+            * ) echo 'Please answer either "yes" or "no".';;
+        esac
+    done
```

Comment:
It looks like some code is duplicated. Is it possible to have a common utils.sh?

Reply:
As discussed in person, I'd like to do this eventually, but don't want to make large changes in this area until the UI plugin automation project is complete.